docs/enhancements/0002_day0-integration.md

```diff
@@ -0,0 +1,282 @@
+to install the cluster and the container image will be used to notify
+[MCO](https://docs.openshift.com/container-platform/4.14/post_installation_configuration/machine-configuration-tasks.html#understanding-the-machine-config-operator)
+that the image on the nodes is a custom image.
+
+When a customers has installed a "day-0 cluster" as mentioned above, we can
+assume that there is a `MachineConfig` in the cluster that looks something like
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-ybettan-external-image
+spec:
+  osImageURL: quay.io/ybettan/rhcos:414.92.202312311229-0
+```
+The important piece in it is `osImageURL: quay.io/ybettan/rhcos:414.92.202312311229-0`
+which tells MCO that the ostree-commit that is supposed to be running on
+the node is encapsulated in `quay.io/ybettan/rhcos:414.92.202312311229-0`.
+
+If MCO finds out that the OS on the node is different than the image specified
+in `osImageURL` then it will override the node OS based on the `MachineConfig`
+and reboot - we are going to leverage this mechanism for the integration with
+KMM.
+
+In a day-0-cluster scenario, the user has applied this `MachineConfig` to tell
+MCO he has modified the OS on the nodes and that it shoudn't override it with
+the RHCOS image from the payload.
+
+### Setting up the integration
+
+A user that wishes to hand over the management of his day-0 kmod to KMM will
+first need to make sure the `MachineConfig` can be reconize by KMM, therefore,
+he would add a `machineconfiguration.openshift.io/kmm-managed: "true"` label to
+it.
+
+KMM will watch for `MachineConfig`s in the cluster containing the
+`machineconfiguration.openshift.io/kmm-managed: "true"` label and create a new
+`ModuleDay0` CR in the cluster for each one of them.
+We will have a different controller for this new CRD in KMM.
+**There should be no more than 1 `ModuleDay0` targeting a `MachineConfigPool`
+(masters/workers).
+If multiple kmods are required they should all be backed in the same image**
+
+Since MCO can only apply `MachineConfig`s changes to an entire `MachineConfigPool`,
+we must require that a `ModuleDay0` will target a `MachineConfigPool` as well.
+MCO ties between the `machineconfiguration.openshift.io/role: worker` label on
+the `MachineConfig` and the `node-role.kubernetes.io/worker: ""` label on the
+node.
+
+A customer can always create his own `MachineConfigPool` and add only 1 worker
+to it but I don't think this should be one of our primary concerns ATM for the
+following reasons
+  * assisted-installer will generate the same ISO for all nodes in the cluster
+    anyway so there is no benefit of having finer grained node selector for
+    `ModuleDay0` when assisted doesn't support such resolution on its own.
+  * if a user wishes to have only some worker nodes modified, he will need to
```

Comment:
can u count how many reboots that cluster will go through untill it stabilizes to the target deployment?

Reply:
numReboots=numWorkers-numNodesInTheNewPool

Anyway, I have removed this section from the proposal at it is MCO's internal implementation and not really related to the proposal